src/map/itemdb.cpp

```diff
@@ -2274,194 +2272,396 @@ bool itemdb_isNoEquip(struct item_data *id, uint16 m) {
+		std::string name;
 
-		lines++;
+		if (!this->asString(node, "Option", name))
+			return 0;
 
-		if (line[0] == '/' && line[1] == '/') // Ignore comments
-			continue;
+		if (random_option_db.option_search_name(name)) {
+			this->invalidWarning(node["Option"], "Found duplicate random option name for %s, skipping.\n", name.c_str());
+			return 0;
+		}
 
-		memset(str, 0, sizeof(str));
+		randopt->name = name;
+	}
 
-		p = line;
+	if (this->nodeExists(node, "Script")) {
+		std::string script;
 
-		p = trim(p);
+		if (!this->asString(node, "Script", script))
+			return 0;
 
-		if (*p == '\0')
-			continue;// empty line
+		if (randopt->script) {
+			aFree(randopt->script);
+			randopt->script = nullptr;
+		}
 
-		if (!strchr(p, ','))
-		{
-			ShowError("itemdb_read_randomopt: Insufficient columns in line %d of \"%s\", skipping.\n", lines, path);
-			continue;
+		randopt->script = parse_script(script.c_str(), this->getCurrentFile().c_str(), id, SCRIPT_IGNORE_EXTERNAL_BRACKETS);
+	}
+
+	if (!exists)
+		this->put(id, randopt);
+
+	return 1;
+}
+
+RandomOptionDatabase random_option_db;
+
+/**
+ * Check if the given random option name exists.
+ * @param name: Random option name
+ * @return True on success or false on failure
+ */
+bool RandomOptionDatabase::option_search_name(std::string name) {
+	for (const auto &opt : random_option_db) {
+		if (opt.second->name.compare(name) == 0)
+			return true;
+	}
+
+	return false;
+}
+
+/**
+ * Return the constant value of the given random option.
+ * @param name: Random option name
+ * @param id: Random option ID
+ * @return True on success or false on failure
+ */
+bool RandomOptionDatabase::option_get_id(std::string name, uint16 *id) {
+	for (const auto &opt : random_option_db) {
+		if (opt.second->name.compare(name) == 0) {
+			id[0] = opt.first;
+			return true;
 		}
+	}
 
-		str[0] = p;
-		p = strchr(p, ',');
-		*p = '\0';
-		p++;
+	return false;
+}
 
-		str[1] = p;
+const std::string RandomOptionGroupDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/item_randomopt_group.yml";
+}
 
-		if (str[1][0] != '{') {
-			ShowError("itemdb_read_randomopt(#1): Invalid format (Script column) in line %d of \"%s\", skipping.\n", lines, path);
-			continue;
+/**
+ * Reads and parses an entry from the item_randomopt_group.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
+ */
+uint64 RandomOptionGroupDatabase::parseBodyNode(const YAML::Node &node) {
+	uint16 id;
+
+	if (!this->asUInt16(node, "Id", id))
+		return 0;
+
+	std::shared_ptr<s_random_opt_group> randopt = this->find(id);
+	bool exists = randopt != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "Group", "Slots" }))
+			return 0;
+
+		randopt = std::make_shared<s_random_opt_group>();
+		randopt->id = id;
+	}
+
+	if (this->nodeExists(node, "Group")) {
+		std::string name;
+
+		if (!this->asString(node, "Group", name))
+			return 0;
+
+		if (random_option_group.option_search_name(name)) {
```

Comment:
Only trigger this, if the found id is not the same as the current one, otherwise it might be someone has redefined the group name in import.